packages/vite/src/node/env.ts

```diff
@@ -35,35 +35,38 @@ export function loadEnv(
     }
   }
 
-  for (const file of envFiles) {
-    const path = lookupFile(envDir, [file], { pathOnly: true, rootDir: envDir })
-    if (path) {
-      const parsed = dotenv.parse(fs.readFileSync(path), {
-        debug: process.env.DEBUG?.includes('vite:dotenv') || undefined
+  const parsed = Object.fromEntries(
+    envFiles.flatMap((file) => {
+      const path = lookupFile(envDir, [file], {
+        pathOnly: true,
+        rootDir: envDir
       })
+      if (!path) return []
+      return Object.entries(
+        dotenv.parse(fs.readFileSync(path), {
+          debug: process.env.DEBUG?.includes('vite:dotenv')
+        })
+      )
+    })
+  )
```

Comment:
Nit: Maybe it's me but I find this a bit hard to read 😬 I think it can be represented as:

```suggestion
  const parsed: Record<string, string> = {}

  for (const file of envFiles) {
    const path = lookupFile(envDir, [file], {
      pathOnly: true,
      rootDir: envDir
    })
    if (path) {
      Object.assign(
        parsed,
        dotenv.parse(fs.readFileSync(path), {
          debug: process.env.DEBUG?.includes('vite:dotenv')
        })
      )
    }
  }
```

Reply:
I just realized @sapphi-red suggested this 😅 I think it's a matter of taste. I don't have a hard preference for this suggestion.